notebooks/08A_composing_parallel_sorting_data.ipynb

```diff
@@ -281,23 +295,27 @@
     "        print(f\"Skipping {key} ({len(ttk.profile)} profiles) because it has duplicate num_procs\")\n",
     "        rm_profs += ttk.profile   \n",
     "tk = tk.filter_profile([p for p in tk.profile if p not in set(rm_profs)])\n",
-    "print(f\"Total files after removing duplicates: {len(tk.profile)}\")"
+    "print(f\"Total profiles after removing duplicates: {len(tk.profile)}\")"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# 3. Create Features"
+    "## 4. Create Features\n",
+    "\n",
+    "In this section, we structure the performance data where each column is a feature, and each row is a feature vector for one performance profile, which is necessary for modeling"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# 3A. Query the Call Tree\n",
+    "### 4A. Query the Call Tree and Re-compose the Thicket\n",
     "\n",
-    "To get performance metrics per node."
+    "Here we select only the nodes we want to compare for our modeling using `Thicket.query()` and `Thicket.concat_thickets()`.\n",
```

Comment:
This is the hardest part to explain. Here's an example:

This is the tree before step 4
```
0.001 correctness_check
0.000 data_init
nan main
├─ nan MPI_Barrier
├─ nan comm
│  ├─ nan MPI_Barrier
│  ├─ nan comm_barrier
│  ├─ nan comm_large
│  │  ├─ nan MPI_Alltoall
│  │  ├─ nan MPI_Alltoallv
│  │  ├─ nan MPI_Bcast
│  │  ├─ nan MPI_Gather
│  │  ├─ nan MPI_Gatherv
│  │  ├─ nan MPI_Isend
│  │  ├─ nan MPI_Recv
│  │  ├─ nan MPI_Scatter
│  │  ├─ nan MPI_Send
│  │  ├─ nan MPI_Sendrecv
│  │  ├─ nan comm_small
│  │  ├─ nan mpi_gather
│  │  └─ nan mpi_scatter
│  ├─ nan comm_large_gather
│  ├─ nan comm_large_scatter
│  └─ nan comm_small
│     ├─ nan MPI_Allgather
│     ├─ nan MPI_Allreduce
│     ├─ nan MPI_Alltoall
│     ├─ nan MPI_Bcast
│     ├─ nan MPI_Gather
│     ├─ nan MPI_Isend
│     ├─ nan MPI_Recv
│     ├─ nan MPI_Reduce
│     ├─ nan MPI_Send
│     └─ nan MPI_Sendrecv
├─ nan comm_large
├─ nan comm_small
├─ nan comp
│  ├─ nan comm
│  │  ├─ nan comm_large
│  │  │  └─ nan Gatherv
│  │  ├─ nan comm_small
│  │  │  ├─ nan Allgather
│  │  │  └─ nan Gather
│  │  ├─ nan comp_large
│  │  └─ nan comp_small
│  ├─ nan comp_large
│  │  ├─ nan comm
│  │  │  └─ nan comm_large
│  │  ├─ nan comp_small
│  │  └─ nan local_sort
│  └─ nan comp_small
├─ nan comp_large
├─ nan comp_small
├─ nan correct_check
├─ nan correctness_check
├─ nan data_init
├─ nan data_initialization
├─ nan data_validation
├─ nan gatherv
├─ nan main
│  ├─ nan comm
│  │  ├─ nan comm_large
│  │  └─ nan comm_small
│  ├─ nan comp
│  │  ├─ nan comp_large
│  │  └─ nan comp_small
│  ├─ nan correct_check
│  ├─ nan data_init
│  └─ nan mpibarrier
├─ nan mpi_recv
├─ nan mpi_send
├─ nan mpibarrier
├─ nan scatter
├─ nan scatterv
└─ nan whole_computation
   ├─ nan check_correctness
   ├─ nan comm
   │  ├─ nan comm_gather
   │  ├─ nan comm_large
   │  │  ├─ nan MPI_Gather
   │  │  ├─ nan MPI_Recv
   │  │  ├─ nan MPI_Scatter
   │  │  └─ nan MPI_Send
   │  ├─ nan comm_scatter
   │  └─ nan comp_large
   │     └─ nan comm_large
   ├─ nan comp
   │  ├─ nan comp_large
   │  │  └─ nan comm_large
   │  └─ nan comp_small
   └─ nan data_init
0.016 whole_comp_time
├─ 0.011 comm
├─ nan comm_large
├─ nan comm_small
├─ 0.005 comp
├─ nan comp_large
└─ nan comp_small
nan whole_computation
├─ nan comm
│  ├─ nan comm_large
│  └─ nan comm_small
├─ nan comp
│  ├─ nan comp_large
│  └─ nan comp_small
├─ nan correctness_check
└─ nan data_init
```
This is the tree after 4A
```
0.072 comm
0.072 comm_large
nan comm_small
0.062 comp
0.062 comp_large
nan comp_small
```
As a result of creating the generalized tree and enforcing annotations from it, the parent/child relationships in the graph don't matter anymore. So in a sense we don't even need the graph we just need to select the nodes from the generalized tree that we are going to use.

Reply:
f6c13c3c should help with the clarity issue since the query and composition do not have to be explained in the same cell and are different operations.

Reply:
27f2074f I took a stab at adding context to the query, but I am not sure exactly how to explain this to a user.